test/unit/host_test.rb

```diff
@@ -1123,6 +1123,19 @@ def teardown
       assert_kind_of Nic::Managed, host.interfaces.find_by_identifier('eth1')
     end
 
+    test "host can't have more interfaces with same identifier" do
+      host = FactoryGirl.build(:host, :managed)
+      host.primary_interface.identifier = 'eth0'
+      nic = host.interfaces.build(:identifier => 'eth0')
+      refute host.valid?
+      assert_present nic.errors[:identifier]
+      assert_present host.errors[:interfaces]
+      nic.identifier = 'eth1'
+      host.valid?
```

Comment:
ok, I get it.
in rails 4 this should be changed to host.validate! so it won't be an open question, it mainly disturbs me grammatically :smile: